lib/puppet/util/windows/principal.rb

```diff
@@ -146,6 +147,12 @@ def self.lookup_account_sid(system_name = nil, sid_bytes)
       end
     end
 
+    # Sanitize the given account name for lookup to avoid known issues
+    def self.sanitize_account_name(account_name)
+      return account_name unless account_name.start_with?('APPLICATION PACKAGE AUTHORITY\\')
+      account_name.split('\\').last
+    end
```

Comment:
since this is a helper function, and it's scope it is to be used in this class, i would make this private

Reply:
Good point. I've also removed: